unison-cli/src/Unison/LSP/Queries.hs

```diff
@@ -5,27 +5,76 @@ module Unison.LSP.Queries
     findSmallestEnclosingNode,
     findSmallestEnclosingType,
     refInDecl,
+    getTypeOfReferent,
+    getTypeDeclaration,
   )
 where
 
 import Control.Lens
+import Control.Monad.Reader
+import Language.LSP.Types
 import qualified Unison.ABT as ABT
+import qualified Unison.Codebase as Codebase
+import Unison.ConstructorReference (GConstructorReference (..))
 import qualified Unison.ConstructorType as CT
+import Unison.DataDeclaration (Decl)
 import qualified Unison.DataDeclaration as DD
+import Unison.LSP.FileAnalysis
 import Unison.LSP.Orphans ()
+import Unison.LSP.Types
 import Unison.LabeledDependency
 import qualified Unison.LabeledDependency as LD
 import Unison.Lexer.Pos (Pos (..))
 import Unison.Parser.Ann (Ann)
 import qualified Unison.Parser.Ann as Ann
 import Unison.Prelude
 import Unison.Reference (TypeReference)
+import qualified Unison.Reference as Reference
+import Unison.Referent (Referent)
+import qualified Unison.Referent as Referent
 import Unison.Symbol (Symbol)
 import Unison.Term (MatchCase (MatchCase), Term)
 import qualified Unison.Term as Term
 import Unison.Type (Type)
 import qualified Unison.Type as Type
 
+-- | Gets the type of a reference from either the parsed file or the codebase.
+getTypeOfReferent :: Uri -> Referent -> MaybeT Lsp (Type Symbol Ann)
```

Comment:
These combinators behave like the codebase ones, but will also find things that are ONLY in the scratch file and haven't been added yet.